src/compile/data/parse.ts

```diff
@@ -396,9 +396,6 @@ export function parseData(model: Model): DataComponent {
   if (isFacetModel(model)) {
     const facetName = model.getName('facet');
 
-    // Derive new sort index field for facet's sort array
-    head = CalculateNode.parseAllForSortIndex(head, model);
```

Comment:
is this always wrong?

Reply:
We add it above since already. I think it is.